renaissance-core/src/main/java/org/renaissance/Benchmark.java

```diff
@@ -0,0 +1,90 @@
+package org.renaissance;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+public interface Benchmark {
+
+  /**
+   * Name of the benchmark. Defaults to kebab-case variant of the benchmark's
+   * class name (without the package prefix). For example, the default name for
+   * a class named {@code LittleBigBench} will be {@code little-big-bench}.
+   */
+  @Documented
+  @Retention(RetentionPolicy.RUNTIME)
+  @interface Name {
+
+    String value();
+
+  }
```

Comment:
Should we ever have anything other than the default? IMO, configurability of the name could lead to confusions.

Reply:
As noted elsewhere, I would leave it there. I would actually prefer those annotations to be set explicitly -- they are logical things, whereas implementation class and subprojects are implementation and build things. 

Consider for example searching through the codebase for "fj-kmeans" -- you will not find it there. Maybe by accident. But if a benchmark has an annotation `@Name("fj-kmeans")` set, it will be spotted immediately. Also consider that me might get to a point where we will have more subprojects to separate dependencies, but the benchmarks in two or more subprojects will be a logical group.

I don't like implicit things, because they require context. I think it a bad practice for implementation things to serve as implicits for logical things that should be decided by authors (subject to approval).

Reply:
Maybe we should then mandate the `@Name` annotation, *and* mandate that it corresponds to the kebab-case version of the class' name.
I.e. I agree that it's good to avoid implicit-ness, but I would also avoid against too many axes of configuration here.

Reply:
I used explicit annotations and will add a build check.